segments/weather.sh

```diff
@@ -177,23 +179,38 @@ __get_yrno_condition_symbol() {
 	esac
 }
 
-__read_tmp_file() {
-	if [ ! -f "$tmp_file" ]; then
+__read_file_split() {
```

Comment:
Nice refactor, now this functinality can be reused.

Unfortuantely in bash we dont' have named arguments for functions with the benefit of documenting the function interface. What I like to do instead is to start each function by declaring local variables with explaining names as a way to document the interface, and subsequently make the function easier to understand with clear variable names.

Thus I would suggest to add this 

```suggestion
__read_file_split() {
	local file="$1"
	local lineno="$2"
	local errmsg="$3"
	[...]
```